External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -47,6 +47,19 @@ uint64_t getMask(IROp_Header* Op) {
   return (~0ULL) >> (64 - NumBits);
 }
 
+// Returns true if the number bits from [0:width) contain the same bit.
+// Ensuring that the consecutive bits in the range are entirely 0 or 1.
+static bool HasConsecutiveBits(uint64_t imm, unsigned width) {
+  uint64_t StartingBit = imm & 1;
+  for (size_t i = 1; i < width; ++i) {
+    if (((imm >> i) & 1) != StartingBit) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
No need to optimise this, but if you wanted to, it could look like:

```c
  // assumes width <= 64
  if (width == 0) return true;
  return ((imm ^ (imm >> 1)) & (((uint64_t)1 << (width-1)) - 1)) == 0;
```

https://godbolt.org/z/PdWdx7zna

Reply:
Very clean implementation! Switched out to your implementation.
Quite a bit nicer, thanks!